metadata-ingestion/src/datahub/ingestion/source/tableau/tableau.py

```diff
@@ -2224,10 +2234,9 @@ def emit_datasource(
         # Tags
         if datasource_info:
             tags = self.get_tags(datasource_info)
-            if tags:
```

Comment:
IMO, `if tags:` block is still needed, isn't it?

Reply:
but tags could be empty, isn't it ? `make_global_tag_aspect_with_tag_list` seems not to deal with empty array.

Reply:
Yes, exactly. If the tags are empty (empty array) we should IMO still save it to properly handle the removal of all tags. Imagine there were tags before and all of them got removed in Tableau. They should also be removed in Datahub, right? Or am I misunderstanding something?

Reply:
This makes sense however it has potential of overwriting tags set via UI which weren't overwritten before for the assets without tags. I see config `ingest_tags` is disabled by default and already mentions that this will override Tags entered from UI so this behavior is fine if `ingest_tags` is enabled. 

However we should not ingest empty globalTags if `ingest_tags` is disabled. I believe later case needs to be fixed in this PR.

Reply:
Also `tags_for_hidden_assets` should probably work only if `ingest_tags` is enabled.